databroker/src/grpc/kuksa_val_v1/val.rs

```diff
@@ -344,12 +283,146 @@ impl proto::val_server::Val for broker::DataBroker {
         }))
     }
 
+    type StreamedUpdateStream = ReceiverStream<Result<proto::StreamedUpdateResponse, tonic::Status>>;
+
+    async fn streamed_update(
+        &self,
+        request: tonic::Request<Streaming<proto::StreamedUpdateRequest>>,
+    ) -> Result<tonic::Response<Self::StreamedUpdateStream>, tonic::Status> {
+        debug!(?request);
+        let permissions = match request.extensions().get::<Permissions>() {
+            Some(permissions) => {
+                debug!(?permissions);
+                permissions.clone()
+            }
+            None => return Err(tonic::Status::unauthenticated("Unauthenticated")),
+        };
+        let mut stream = request.into_inner();
+
+        let mut shutdown_trigger = self.get_shutdown_trigger();
+
+        // Copy (to move into task below)
+        let broker = self.clone();
+
+        // Create stream (to be returned)
+        let (sender, receiver) = mpsc::channel(10);
+        // Listening on stream
+        tokio::spawn(async move {
+            let permissions = permissions;
+            let broker = broker.authorized_access(&permissions);
+            loop {
```

Comment:
I am no Rust expert - anything that prevents dividing it into multiple methods to limit length and depth?

Reply:
By now it is possible to extract most of the loop into a separate method:
```
async fn handle_streamed_update_message(
    broker: &AuthorizedAccess,
    sender: &Sender<Result<StreamedUpdateResponse, Status>>,
    message: Result<Option<StreamedUpdateRequest>, Status>,
) {
```
However, here I'm facing some Rust specific issues again:
```
error[E0597]: `broker` does not live long enough
let broker = broker.authorized_access(&permissions);
argument requires that `broker` is borrowed for `'static`
```
and
```
error[E0597]: `permissions` does not live long enough
let broker = broker.authorized_access(&permissions);
argument requires that `permissions` is borrowed for `'static`
```

I don't easily want to make this change, because it would keep the instance alive as long as the application is running if I understood this correctly. 

So probably this can be fixed, however I don't know yet how the fix would be made appropriate.